kuksa_apps/kanto/edge_client.py

```diff
@@ -0,0 +1,122 @@
+# Copyright (c) 2022 Contributors to the Eclipse Foundation
+#
+# See the NOTICE file(s) distributed with this work for additional
+# information regarding copyright ownership.
+#
+# This program and the accompanying materials are made available under the
+# terms of the Eclipse Public License 2.0 which is available at
+# http://www.eclipse.org/legal/epl-2.0
+#
+# SPDX-License-Identifier: EPL-2.0
+
+import signal
+import sys
+
+import paho.mqtt.client as mqtt
+from ditto.client import Client
+from ditto.model.feature import Feature
+from ditto.protocol.things.commands import Command
+from kuksa_viss_client import KuksaClientThread
+
+from edge_device_info import EdgeDeviceInfo
+from utils import process_tree, process_signal
+
+FEATURE_ID_VSS = "VSS"
+
+EDGE_CLOUD_CONNECTOR_TOPIC_DEV_INFO = "edge/thing/response"
+EDGE_CLOUD_CONNECTOR_TOPIC_DEV_INFO_REQUEST = "edge/thing/request"
+
+
+class EdgeClient:
+    def __init__(self):
+        self.mqtt_client = None
+        cfg = {"ip": "127.0.0.1"}
+        self.kuksa_client = KuksaClientThread(cfg)
+        self.device_info = None
+        self.ditto_client = None
+
+    def on_connect(self, client:mqtt.Client, obj, flags, rc):
+        print("Connected with result code " + str(rc))
+        self.mqtt_client = client
+        # init ditto client
+        self.ditto_client = Client(paho_client=self.mqtt_client)
+        self.ditto_client.connect()
+        # init kuksa client
+        self.kuksa_client.start()
+        self.kuksa_client.authorize()
+        # trigger initialization
+        self.mqtt_client.subscribe(EDGE_CLOUD_CONNECTOR_TOPIC_DEV_INFO)
+        self.mqtt_client.publish(EDGE_CLOUD_CONNECTOR_TOPIC_DEV_INFO_REQUEST, None, 1)
+
+    def on_message(self, client, userdata, msg):
+        try:
+            if msg.topic == EDGE_CLOUD_CONNECTOR_TOPIC_DEV_INFO:
+                if self.device_info is None:
+                    self.device_info = EdgeDeviceInfo()
+                    self.device_info.unmarshal_json(msg.payload)
+                    self.add_vss_feature()
+                    self.subscribe()
+                else:
+                    print('device info already available - discarding message')
+                return
+        except Exception as ex:
+            print(ex)
+
+    def subscribe(self):
+        self.kuksa_client.subscribe('Vehicle.Cabin.Infotainment.Navigation.CurrentLocation.Altitude', self.on_kuksa_signal)
```

Comment:
As of today this file is hardcoded for `CurrentLocation`. So I assume this file typically needs to be adapted for any project using it. Would it be an idea to include comments to help anyone integrating/refactoring this app, e.g. that they (most likely) needs to change the lines below?